wiki/src/main/resources/XWiki/PDFViewerMacro.xml

```diff
@@ -174,25 +174,39 @@
 {{velocity output="false"}}
 #macro (getUrlFromFile $file)
   #set ($url = $NULL)
+  ## If the url is not directly specified, the attachment reference can be taken either directly from the file macro
+  ## parameter for xwiki versions starting with 11.5, since an attachment picker is used, or by using both the file and
+  ## document parameters.
   #if ($file.startsWith('http://') || $file.startsWith('https://'))
     #set ($url = $file)
   #else
+    ## For backwards compatibility with the macros added in older versions, first it is checked if the file name
+    ## specified exists on the defined document or on the current one before considering the file parameter as the full
+    ## reference. The filename is needed for the error message when no attachment is found.
+    #set ($filename = $file)
+    #set ($authorRightsNeeded = false)
     #if ("$!docname" != '')
       #set ($document = $xwiki.getDocument($docname))
+      #if (!$document.getAttachment($file) &amp;&amp; $asAuthor &amp;&amp; $xwiki.hasProgrammingRights() &amp;&amp;
+          $xwiki.exists('XWiki.PDFViewerService'))
+        #set ($document = $xwiki.getDocumentAsAuthor($docname))
+        #set ($authorRightsNeeded = true)
+      #end
```

Comment:
Shouldn't this be checked also when the attachment is loaded from the current document, or when the attachment is specified using a reference? It's not clear to me what we do this only when the source document is specified using the macro parameter.

Reply:
For when the attachment is specified using a reference I missed the case.
But I am not sure if this should also be done for when the attachment is loaded from the current document. If I can see the page that has the macro, it means that I have view rights on the page and I would have the rights for the attachment as well, isn't it? I am not sure if there can be a case were I cannot get the attachment

Reply:
Yes, I think you can skip this for the current document. Note that the condition won't apply to the current document anyway because ``$document.getAttachment($file)`` should always return the attachment if it exists, so technically you could still apply this to all cases, knowing that it won't enter for the current document. I would choose whatever is the easiest to write :)